lib/experimental/class-wp-webfonts.php

```diff
@@ -44,192 +12,177 @@ class WP_Webfonts {
+	 * Registers a variation for a font family.
 	 *
-	 * @param string $font_family_name The font family name to be enqueued.
-	 * @return bool True if successfully enqueued, else false.
+	 * @param string $font_family
+	 * @param string $variation_handle
+	 * @param array  $variation
 	 */
-	public function enqueue_webfont( $font_family_name ) {
-		$slug = $this->get_font_slug( $font_family_name );
+	public function add_variation( $font_family, $variation_handle, $variation ) {
+		$font_family_handle = sanitize_title( $font_family );
 
-		if ( isset( $this->enqueued_webfonts[ $slug ] ) ) {
-			return true;
+		// Register the font family when it does not yet exist.
+		if ( ! isset( $this->registered[ $font_family ] ) ) {
+			$this->add( $font_family_handle, false );
 		}
 
-		if ( ! isset( $this->registered_webfonts[ $slug ] ) ) {
-			/* translators: %s unique slug to identify the font family of the webfont */
-			_doing_it_wrong( __METHOD__, sprintf( __( 'The "%s" font family is not registered.', 'gutenberg' ), $slug ), '6.0.0' );
+		$variation = $this->validate_variation( $font_family, $variation );
 
+		// Variation validation failed.
+		if ( ! $variation ) {
 			return false;
 		}
 
-		$this->enqueued_webfonts[ $slug ] = $this->registered_webfonts[ $slug ];
-		unset( $this->registered_webfonts[ $slug ] );
-		return true;
+		if ( $variation['src'] ) {
+			$result = $this->add( $variation_handle, $variation['src'], array(), false, array( 'font-properties' => $variation ) );
+		} else {
+			$result = $this->add( $variation_handle, false, array(), false, array( 'font-properties' => $variation ) );
+		}
+
+		if ( $result ) {
+			$this->providers[ $variation['provider'] ]['fonts'][] = $variation_handle;
+		}
+
+		return $result;
 	}
 
 	/**
-	 * Get the font slug.
-	 *
-	 * @since 6.0.0
+	 * Adds a variation as a dependency for the main font alias.
 	 *
-	 * @param array|string $to_convert The value to convert into a slug. Expected as the web font's array
-	 *                                 or a font-family as a string.
-	 * @return string|false The font slug on success, or false if the font-family cannot be determined.
+	 * @param $font_family_handle
+	 * @param $variation_handle
 	 */
-	public static function get_font_slug( $to_convert ) {
-		if ( is_array( $to_convert ) ) {
-			if ( isset( $to_convert['font-family'] ) ) {
-				$to_convert = $to_convert['font-family'];
-			} elseif ( isset( $to_convert['fontFamily'] ) ) {
-				$to_convert = $to_convert['fontFamily'];
-			} else {
-				_doing_it_wrong( __METHOD__, __( 'Could not determine the font family name.', 'gutenberg' ), '6.0.0' );
-				return false;
-			}
-		}
-
-		return sanitize_title( $to_convert );
+	public function add_dependency( $font_family_handle, $variation_handle ) {
+		$dependencies = $this->registered[ $font_family_handle ];
+		$dependencies[] = $variation_handle;
+		$this->registered[ $font_family_handle ] = $dependencies;
```

Comment:
Is there a reason this can't be simplified to this?

```suggestion
		$this->registered[ $font_family_handle ][] = $variation_handle;
```

Reply:
Missed this initially! I actually had a typo here and was not properly referencing the dependencies correctly. I fixed the typo and applied this simplification.